tools/checked-c-convert/ProgramInfo.cpp

```diff
@@ -100,15 +115,26 @@ PointerVariableConstraint::mkString(Constraints::EnvironmentMap &E) {
     ConstAtom *C = E[&VA];
     assert(C != nullptr);
 
+    std::map<uint32_t, Qualification>::iterator q;
     switch (C->getKind()) {
     case Atom::A_Ptr:
+      q = QualMap.find(V);
+      if (q != QualMap.end())
+        if (q->second == ConstQualification)
+          s = s + "const ";
+
       emittedBase = false;
       s = s + "_Ptr<";
 
       caratsToAdd++;
       break;
     case Atom::A_Arr:
     case Atom::A_Wild:
+      q = QualMap.find(V);
```

Comment:
Ah, I think that what should happen here is that the insertion of `const` based on `q` should occur after we print out everything, not before. If the value pointed to is `const` then that should be captured by `BaseType`. I've made that change.